torch/_dynamo/guards.py

```diff
@@ -190,7 +191,10 @@ def __init__(
         self.id_ref = id_ref
         self.source_ref = source_ref
         self.lookup_weakrefs = lookup_weakrefs
-        self.scope: Dict[str, Dict[str, object]] = {"L": local_scope, "G": global_scope}
+        self.scope: Dict[str, Dict[str, object]] = {
+            "L": {k: unwrap_if_wrapper(v) for k, v in local_scope.items()},
+            "G": {k: unwrap_if_wrapper(v) for k, v in global_scope.items()},
```

Comment:
This wont work because you are making a copy of f_globals. So, any change in f_globals later on will not be tracked.